bundles/org.openhab.binding.heliosventilation/src/main/java/org/openhab/binding/heliosventilation/internal/HeliosVentilationDataPoint.java

```diff
@@ -0,0 +1,310 @@
+     *
+     * @return the name of the variable, which is also the channel name
+     */
+    public String getName() {
+        return name;
+    }
+
+    /**
+     *
+     * @return address of the variable
+     */
+    public byte address() {
+        return address;
+    }
+
+    /**
+     * @return the bit mask of the data point. 0xFF in case the full byte is used.
+     */
+    public byte bitMask() {
+        byte mask = (byte) 0xff;
+        if (datatype == DataType.NUMBER || datatype == DataType.SWITCH) {
+            mask = (byte) (((1 << bitLength) - 1) << bitStart);
+        }
+        return mask;
+    }
+
+    /**
+     * interpret the given byte b and return the value as State.
+     *
+     * @param b
+     * @return state representation of byte value b in current datatype
+     */
+    public State asState(byte b) {
+        int val = b & 0xff;
+        if (datatype == DataType.TEMPERATURE) {
+            return new QuantityType<>(TEMP_MAP[val], SIUnits.CELSIUS);
+        } else if (datatype == DataType.BYTE_PERCENT) {
+            return new QuantityType<>((int) ((val - BYTE_PERCENT_OFFSET) * 100.0 / (255 - BYTE_PERCENT_OFFSET)),
+                    SmartHomeUnits.PERCENT);
+        } else if (datatype == DataType.SWITCH && bitLength == 1) {
+            if ((b & (1 << bitStart)) != 0) {
+                return OnOffType.ON;
+            } else {
+                return OnOffType.OFF;
+            }
+        } else if (datatype == DataType.NUMBER) {
+            int value = (b & bitMask()) >> bitStart;
+            return new DecimalType(value);
+        } else if (datatype == DataType.PERCENT) {
+            return new QuantityType<>(val, SmartHomeUnits.PERCENT);
+        } else if (datatype == DataType.FANSPEED) {
+            int i = 1;
+            while (i < FANSPEED_MAP.length && FANSPEED_MAP[i] < val) {
+                i++;
+            }
+            return new DecimalType(i);
+        } else if (datatype == DataType.HYSTERESIS) {
+            return new QuantityType<>(val / 3, SIUnits.CELSIUS);
+        }
+
+        return UnDefType.UNDEF;
+    }
+
+    /**
+     * interpret the given byte b and return the value as string.
+     *
+     * @param b
+     * @return sting representation of byte value b in current datatype
+     */
+    public String asString(byte b) {
+        State ste = asState(b);
+        String str = ste.toString();
+        if (ste instanceof UnDefType) {
+            return String.format("<unknown type> %02X ", b);
+        } else {
+            return str;
+        }
+    }
+
+    /**
+     * generate byte data to transmit
+     *
+     * @param val is the state of a channel
+     * @return byte value with RS485 representation. Bit level values are returned in the correct location, but other
+     *         bits/datapoints in the same address are zero.
+     */
+    public byte getTransmitDataFor(State val) {
+        byte result = 0;
+        DecimalType value = val.as(DecimalType.class);
```

Comment:
I'm not sure that this does a proper conversion if the QuantityType supplied here is of an unexpected unit.

Reply:
With the unexpected unit we anyhow cannot do anything reasonable which is why the if (value == null) block is empty. And due to the thing/item types we should not get anything unexpected, should we?

Do you have a concrete case in mind where this could happen and can you suggest something what to do here?

Reply:
You simply need to convert the quantity type to the value you want:
`((QuantityType<?>) val).toUnit(SIUnits.CELSIUS).doubleValue()`